### DISCUSSION TITLE
How in Cypress, find a card with a specific text and click on a specific button?

### DISCUSSION THREAD

**Discussion by Guid21** [cypress-io/cypress#22615] [Questions and Help] [answered]
Hello, everyone!

Faced with a problem, I want to find the card I need in the list of cards guided by the text, and click in this card on the button

The idea is that would not go into details of nesting

Enclosing a screenshot of how it looks like
![image](https://user-images.githubusercontent.com/16309665/176655183-35e77313-b2ee-4d47-8b52-41d3d473d03f.png)

**html structure:**

```
<div class="card">
  <span>name 1</span>
  <button data-test="more-menu"><svg /></button>
</div>
<div class="card">
  <span>name 2</span>
  <button data-test="more-menu"><svg /></button>
</div>
  <span>name 3</span>
  <button data-test="more-menu"><svg /></button>
</div>
```

**Comment by muhammaduzair98** [comment]
You can use Chai Assertions like `cy.get("Selector").should('have.text','Name 1')`

**Comment by jjhelguero** [comment, marked as answer]
Based on your html, you'll want to find the card by using a selector and its' text by using [`.contains()`](https://docs.cypress.io/api/commands/contains)

```js
cy.contains('.card', 'name 1')
  .find('[data-test="more-menu"]')
  .click()
```

**Reply by TruptiBip** [reply]
It worked for me ..thanks!!!
